From d95a1a573626336a672859b2a5cadca1e0fe8aa1 Mon Sep 17 00:00:00 2001
From: Erlend Eriksen <erlend_ne@hotmail.com>
Date: Mon, 13 Sep 2021 12:16:58 +0200
Subject: [PATCH] assume std allocator traits

---
 roscpp_serialization/include/ros/serialization.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/roscpp_serialization/include/ros/serialization.h b/roscpp_serialization/include/ros/serialization.h
index 7daee97f..271a053e 100644
--- a/roscpp_serialization/include/ros/serialization.h
+++ b/roscpp_serialization/include/ros/serialization.h
@@ -332,7 +332,7 @@ struct VectorSerializer
 template<typename T, class ContainerAllocator>
 struct VectorSerializer<T, ContainerAllocator, typename boost::disable_if<mt::IsFixedSize<T> >::type >
 {
-  typedef std::vector<T, typename ContainerAllocator::template rebind<T>::other> VecType;
+  typedef std::vector<T, typename ContainerAllocator> VecType;
   typedef typename VecType::iterator IteratorType;
   typedef typename VecType::const_iterator ConstIteratorType;
 
@@ -382,7 +382,7 @@ struct VectorSerializer<T, ContainerAllocator, typename boost::disable_if<mt::Is
 template<typename T, class ContainerAllocator>
 struct VectorSerializer<T, ContainerAllocator, typename boost::enable_if<mt::IsSimple<T> >::type >
 {
-  typedef std::vector<T, typename ContainerAllocator::template rebind<T>::other> VecType;
+  typedef std::vector<T, typename ContainerAllocator> VecType;
   typedef typename VecType::iterator IteratorType;
   typedef typename VecType::const_iterator ConstIteratorType;
 
@@ -424,7 +424,7 @@ struct VectorSerializer<T, ContainerAllocator, typename boost::enable_if<mt::IsS
 template<typename T, class ContainerAllocator>
 struct VectorSerializer<T, ContainerAllocator, typename boost::enable_if<mpl::and_<mt::IsFixedSize<T>, mpl::not_<mt::IsSimple<T> > > >::type >
 {
-  typedef std::vector<T, typename ContainerAllocator::template rebind<T>::other> VecType;
+  typedef std::vector<T, typename ContainerAllocator> VecType;
   typedef typename VecType::iterator IteratorType;
   typedef typename VecType::const_iterator ConstIteratorType;
 
-- 
2.32.0

